aerospike-core/src/batch/batch_executor.rs

```diff
@@ -40,73 +40,39 @@ impl BatchExecutor {
         policy: &BatchPolicy,
         batch_reads: Vec<BatchRead>,
     ) -> Result<Vec<BatchRead>> {
-        let mut batch_nodes = self.get_batch_nodes(&batch_reads).await?;
+        let batch_nodes = self.get_batch_nodes(&batch_reads).await?;
         let jobs = batch_nodes
-            .drain()
+            .into_iter()
             .map(|(node, reads)| BatchReadCommand::new(policy, node, reads))
             .collect();
         let reads = self.execute_batch_jobs(jobs, &policy.concurrency).await?;
-        let mut res: Vec<BatchRead> = vec![];
-        for mut read in reads {
-            res.append(&mut read.batch_reads);
-        }
-        Ok(res)
+        let mut all_results: Vec<_> = reads.into_iter().flat_map(|cmd|cmd.batch_reads).collect();
```

Comment:
Important bugfix for batch read. Puts everything back in the order it was asked in.